internals/daemon/daemon.go

```diff
@@ -561,24 +561,13 @@ func (d *Daemon) Stop(sigCh chan<- os.Signal) error {
 		logger.Noticef("Cannot stop running services: %v", err)
 	}
 
-	d.tomb.Kill(nil)
```

Comment:
Kill() is used to signal tomb managed goroutines to exit. However, our http.Server goroutines does not use this directly. Instead, its used for suppressing server errors on exit, which masked part of the original issue in the first place. The daemon tomb is actually used at the top level to trigger a call of this function, so removing this early duplicate (its called again later in this function), makes Serve() errors getting properly reported.

The tomb effectively has two uses:

1. Daemon termination signal using the dying state
2. Waiting for http.Server goroutines to exit cleanly (and supressing errors - which can still happen on a reboot request, which uses the Kill() method - see daemon.Start())

The two are very decoupled right now, so in the future I would be tempted to use a separate channel for termination requests.

Reply:
I decided to undo this change, as it would be very difficult to prove this does not break something. Pebble can be terminated from a signal, in which case, Stop() would enter without the http.Server goroutines set do dying. Under this condition, if any connections fail to close, errors will be seen. Without Dying state, those errors will affect the shutdown logic.